packages/ui/src/components/vuestic-components/va-button-dropdown/VaButtonDropdown.vue

```diff
@@ -1,166 +1,145 @@
+<script lang="ts">
+import Vue from 'vue'
+import { Component, Mixins, Inject } from 'vue-property-decorator'
+import VaDropdown from '../va-dropdown/VaDropdown.vue'
+import VaButton from '../va-button/VaButton.vue'
+import VaButtonGroup from '../va-button-group/VaButtonGroup.vue'
 import { ColorThemeMixin } from '../../../services/ColorThemePlugin'
+import { makeContextablePropsMixin } from '../../context-test/context-provide/ContextPlugin'
+import { SizeMixin } from '../../../mixins/SizeMixin'
 
-export default {
-  name: 'VaButtonDropdown',
-  components: { VaButtonGroup, VaButton, VaDropdown },
-  mixins: [ColorThemeMixin],
-  props: {
-    value: {
-      type: Boolean,
-    },
-    buttonProps: {
-      type: Object,
-      default: () => ({
-        small: false,
-        large: false,
-        flat: false,
-        outline: false,
-      }),
-    },
-    label: {
-      type: String,
-      default: '',
-    },
-    disabled: {
-      type: Boolean,
-    },
-    disableButton: {
-      type: Boolean,
-    },
-    disableDropdown: {
-      type: Boolean,
-    },
-    position: {
-      type: String,
-      default: 'bottom',
-    },
-    icon: {
-      type: String,
-      default: 'expand_more',
-    },
-    openedIcon: {
-      type: String,
-      default: 'expand_less',
-    },
-    split: {
-      type: Boolean,
-    },
-    splitTo: {
-      type: String,
-      default: '',
+const ButtonPropsMixin = makeContextablePropsMixin({
+  value: { type: Boolean },
+  outline: { type: Boolean, default: false },
+  disableButton: { type: Boolean, default: false },
+  disableDropdown: { type: Boolean, default: false },
+  flat: { type: Boolean, default: false },
+  disabled: { type: Boolean, default: false },
+  size: {
+    type: String,
+    default: 'medium',
+    validator: (value: string) => {
+      return ['medium', 'small', 'large'].includes(value)
     },
   },
-  data () {
+  split: { type: Boolean },
+  splitTo: { type: String, default: '' },
+  icon: { type: String, default: 'expand_more' },
+  openedIcon: { type: String, default: 'expand_less' },
+  position: { type: String, default: 'bottom' },
+  label: { type: String },
+})
+
+@Component({
+  name: 'VaButtonDropdown',
+  components: { VaButtonGroup, VaButton, VaDropdown },
+})
+export default class VaButtonDropdown extends Mixins(
+  SizeMixin,
+  ButtonPropsMixin,
+  ColorThemeMixin,
+) {
+  @Inject({
+    default: () => ({}),
+  }) readonly va!: any
+
+  showDropdown = false
+
+  get computedIcon (): string {
+    const propsData: any = this.$options.propsData
+    const resultedIcon = propsData.openedIcon || (propsData.icon ? this.c_icon : this.c_openedIcon)
+    return this.showDropdown ? resultedIcon : this.c_icon
+  }
+
+  get computedClass () {
     return {
-      visible: false,
+      'va-button-dropdown': true,
+      'va-button-dropdown--split': this.c_split,
+      [`va-button-dropdown--${this.c_size}`]: true,
     }
```

Comment:
Split into 2 classes with conditions. Generally we do not want to use string composition for css classes as that complicates search.